commands/template_pull_stack.go

```diff
@@ -0,0 +1,94 @@
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"reflect"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/openfaas/faas-cli/stack"
+	"github.com/spf13/cobra"
+)
+
+var (
+	templateURL string
+)
+
+func init() {
+	templatePullStackCmd.Flags().BoolVar(&overwrite, "overwrite", false, "Overwrite existing templates?")
+	templatePullStackCmd.Flags().BoolVar(&pullDebug, "debug", false, "Enable debug output")
+
+	templatePullCmd.AddCommand(templatePullStackCmd)
+}
+
+var templatePullStackCmd = &cobra.Command{
+	Use:   `stack`,
+	Short: `Downloads templates specified in the function definition yaml file`,
+	Long: `Downloads templates specified in the function yaml file, in the current directory
+	`,
+	Example: `
+  faas-cli template pull stack
+  faas-cli template pull stack -f myfunction.yml
+`,
+	RunE: runTemplatePullStack,
+}
+
+func runTemplatePullStack(cmd *cobra.Command, args []string) error {
+	templatesConfig, err := loadTemplateConfig()
+	if err != nil {
+		return err
+	}
+	templateURLs := getURLsFromConfigTemplatesInfo(templatesConfig)
+	err = pullMultipleTemplates(templateURLs)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadTemplateConfig() (stack.TemplatesConfiguration, error) {
+	templateConfiguration := stack.TemplatesConfiguration{}
+
+	stackConfig, err := readStackConfig()
+	if err != nil {
+		return templateConfiguration, err
+	}
+	return stackConfig.StackConfig.TemlpatesConfig, nil
+}
+
+func readStackConfig() (stack.Configuration, error) {
+	configField := stack.Configuration{}
+
+	configFieldBytes, err := ioutil.ReadFile(yamlFile)
+	if err != nil {
+		return configField, fmt.Errorf("Error while reading files %s", err.Error())
+	}
+	unmarshallErr := yaml.Unmarshal(configFieldBytes, &configField)
+	if unmarshallErr != nil {
+		return configField, fmt.Errorf("Error while reading configuration: %s", err.Error())
+	}
+	if reflect.DeepEqual(configField, stack.Configuration{}) {
+		return configField, fmt.Errorf("Error while reading configuration: unable to locate template field")
+	}
+	return configField, nil
+}
+
+func getURLsFromConfigTemplatesInfo(templateConfig stack.TemplatesConfiguration) map[string]string {
```

Comment:
Well I was thinking, further down when you want to save the template, you would not like to write `golang-http-middleware` or something like that you can name it whatever you want. I was thinking of the times when you want to pull specific template by the name you have saved it, I think this is not implemented ATM, I will extend this also pulling template by name in the config.

Reply:
that is the direction I thought you were going to go.  The template pull implementation will probably need to change in the future because right now it just pulls the entire template repository, instead of a single template.  But that implementation can wait until the "name" is being used

Reply:
This is removed as the templates are now map `name:url` :heavy_check_mark: Also custom name is added with flag `--name`